### ISSUE TITLE
Formattable output prefix.

### ISSUE THREAD

**Issue by bitprophet** [fabric/fabric#163]
### Description

It seems that the output prefix should be a bit more flexible than it currently is. The current format is:

```
[hoststring] Message
```

This should be flexible, meaning both format as well as the ability to turn it off, or make it print one line per host/task combo. (That particular aspect has been spun off into #281.)

An ideal situation would be to use  python's built-in template strings. (or to be very forward thinking format strings, but those require 2.6). The template can be put in evn.output.prefix or similar.  Using the templates it is easy to just define output format based on exposed variables.

These could include:
- hostname
- hoststring
- comand output signifier [1][2]
- time [1] 
- other stuff

It could also include color code formatting.

[1] some variables will need to have a helper variable, e.g. strftime format for time
[2] output signifier could be something simple like a *\* making the output something like:

```
[hostname] Log msg
[hostname] ** output from cmd
```

An example format string (with helpers) could look like:

```
env.output.prefix = "[${hostname} - ${time}] ${cmd_sig}"
env.output.timeformat = "H:M"
env.output.cmd_sig = "**"
```

resulting in:

```
[www1.foo.com - 17:05] Executing hostname
[www1.foo.com - 17:05] ** www1.foo.com
```

---

Originally submitted by **Erich Heine** ([sophacles](https://github.com/sophacles)) on **2010-05-14** at **05:56pm EDT**
### Attachments
- [fabric.patch](https://gist.github.com/20a35ef0f3095026b028)
### Relations
- Related to #164: puts show_prefix
- Related to #281: Make line prefix optional
- Related to #57: Support full logging to file
- Related to #7: Improved prompt detection and passthrough

**Comment by ror6ax** [comment]
Currently `env.output.prefix="[]"` returns AttributeError:
 ```
   env.output.prefix="[]"
  File "/usr/lib/python2.6/site-packages/fabric/utils.py", line 204, in __getattr__
    raise AttributeError(key)
AttributeError: output
```

**Comment by bitprophet** [comment]
That looks like a typo, should be `env.output_prefix` - underscore, not dot :)

**Comment by anhvanngouiowa** [comment]
Is there any way to put the boolean value in fabricic file ?
I tried : output_prefix=False
but it understand False as string 'False'.

**Comment by bitprophet** [comment]
@anhvanngouiowa No, that's a limitation of fabricrc files :( strings only. FWIW version 2.0 will have a much less awful config mechanism allowing this sort of setting to be configured usefully.

**Comment by Constantin07** [comment]
I also need a way to customise the output. When this patch is going to be merged into upstream ?

**Comment by bitprophet** [comment]
Roll this into https://github.com/pyinvoke/invoke/issues/15

**Comment by bitprophet** [comment]
On second thought, just make it more obvious that this would be a nice small feature add for the 1.x line.
